core/src/test/java/org/apache/iceberg/TestTableMetadata.java

```diff
@@ -1451,50 +1457,67 @@ public void testCreateV2MetadataThroughTableProperty() {
         .doesNotContainKey(TableProperties.FORMAT_VERSION);
   }
 
-  @Test
-  public void testReplaceV1MetadataToV2ThroughTableProperty() {
+  private static Stream<Arguments> upgradeTableVersionProvider() {
+    // return a stream of all valid upgrade paths
+    return IntStream.range(1, TableMetadata.SUPPORTED_TABLE_FORMAT_VERSION)
+        .boxed()
+        .flatMap(
+            baseTableVersion ->
+                IntStream.rangeClosed(
+                        baseTableVersion + 1, TableMetadata.SUPPORTED_TABLE_FORMAT_VERSION)
+                    .mapToObj(newTableVersion -> arguments(baseTableVersion, newTableVersion)));
+  }
+
+  @ParameterizedTest
+  @MethodSource("upgradeTableVersionProvider")
+  public void testReplaceMetadataThroughTableProperty(int baseTableVersion, int newTableVersion) {
```

Comment:
```suggestion
  public void testReplaceMetadataThroughTableProperty(int baseFormatVersion, int newFormatVersion) {
```

Reply:
@leangjonathan just double checking since it was marked as resolved but I don't think it was.